app/controllers/api/mentor_discussions_controller.rb

```diff
@@ -39,5 +39,16 @@ def create
       # TODO: Return the discussion here
       head 200
     end
+
+    def mark_as_nothing_to_do
+      discussion = ::Solution::MentorDiscussion.find_by(uuid: params[:id])
+
+      return render_404(:mentor_discussion_not_found) if discussion.blank?
+      return render_403(:mentor_discussion_not_accessible) unless discussion.viewable_by?(current_user)
```

Comment:
Let's put this guard here.

```suggestion
      return render_403(:mentor_discussion_not_accessible) unless discussion.viewable_by?(current_user)
      return render_403(:mentor_discussion_not_accessible) unless discussion.mentor_id == current_user.id
```